tests/test_ote_training.py

```diff
@@ -643,10 +806,80 @@ def __call__(self, data_collector: DataCollector,
-        results = {}
+        score_name, score_value = self._run_ote_nncf_export_evaluation(data_collector, **kwargs)
+        results = {
+                'metrics': {
+                    'accuracy': {
+                        score_name: score_value
+                    }
+                }
+        }
         return results
 
+def get_value_from_dict_by_dot_separated_address(struct, address):
+    def _get(cur_struct, addr):
+        assert isinstance(addr, list)
+        if not addr:
+            return cur_struct
+        assert isinstance(cur_struct, dict)
+        assert addr[0] in cur_struct, \
+                f'Cannot find address {address} in struct {struct}: {addr} is absent in {cur_struct}'
+        return _get(cur_struct[addr[0]], addr[1:])
+
+    return _get(struct, address.split('.'))
+
+def _check_mutual_exclusive_required_keys(struct, keys):
+    assert isinstance(struct, dict)
+    assert any(k in struct for k in keys), f'All of the keys {keys} are absent in the struct {struct}'
+    num_occurences = sum(int(k in struct) for k in keys)
+    assert num_occurences == 1, f'Wrong num occurences {num_occurences} of keys {keys} in the dict {struct}'
+
+def validate_test_metrics(current_result, test_results_storage, current_test_expected_metrics):
+    if current_test_expected_metrics == SKIP_VALIDATION():
+        logger.info(f'current_test_expected_metrics is SKIP_VALIDATION, validation of test results is skipped')
+        return
+    if current_test_expected_metrics == ABSENT_EXPECTED_VALIDATION_METRICS():
+        raise RuntimeError(f'Validation should be run, but expected metrics for the current test are absent')
+
+    assert isinstance(current_test_expected_metrics, dict), \
+            f'Wrong current test expected metric: {current_test_expected_metrics}'
+    is_passed = True
+    fail_reasons = []
+    logger.info('Validation: begin')
+    for k, v in current_test_expected_metrics.items():
+        cur_res_addr = k
+        cur_metric_requirements = v
+        logger.info(f'Validation of test results: begin check {cur_res_addr}')
+        current_metric = get_value_from_dict_by_dot_separated_address(current_result, cur_res_addr)
+        _check_mutual_exclusive_required_keys(cur_metric_requirements, ['target_value', 'base'])
+        if 'target_value' in cur_metric_requirements:
+            target_value = float(cur_metric_requirements['target_value'])
+        elif 'base' in cur_metric_requirements:
+            base_metric_address = cur_metric_requirements['base']
+            target_value = get_value_from_dict_by_dot_separated_address(test_results_storage, base_metric_address)
+            target_value = float(target_value)
+        else:
+            raise RuntimeError(f'Wrong expected metrics {current_test_expected_metrics}')
+
+        if 'max_drop' in cur_metric_requirements:
+            target_value = target_value - float(cur_metric_requirements['max_drop'])
```

Comment:
No, usually the more is the better.
Note that when we measure time of stage execution in the future (that is "the less the better"), we will require to point direction "what is better -- more or less" in the YAML file (as in accuracy checker).
But I'd like to finish this PR as is now.

Reply:
It might be the common way to compare the difference for NNCF/POT compression. Let's @alexsu52 @AlexanderDokuchaev better comment on this. But for ONNX/OpenVINO export we are not expecting the metric to become higher that it was for original model. If, given a limited amount of data, we end up seeing much larger metric values for the exported model then most likely we do have a bug.

Reply:
I agree with @druzhkov-paul. This statement is valid for the POT and for the NNCF export and validation tasks. We can face the issue with NNCF because we can stop optimization only at the end of an epoch training. My general proposal is to use `abs` for all tasks except the NNCF optimization task. Probably, the comparator type needs to add to the config.